python/pyarrow/_dataset.pyx

```diff
@@ -186,6 +207,23 @@ cdef class FileSystemDiscoveryOptions:
     cdef inline CFileSystemDiscoveryOptions unwrap(self):
         return self.options
 
+    @property
+    def partition_scheme(self):
+        cdef shared_ptr[CPartitionScheme] s = self.options.partition_scheme.scheme()
+        if s.get() != nullptr:
+            return PartitionScheme.wrap(s)
+        return PartitionSchemeDiscovery.wrap(self.options.partition_scheme.discovery())
```

Comment:
Return `None` and add a `partition_scheme_discovery` method.

Reply:
will do